### PR TITLE
fix(notification): update status icon description prop

### LINKED ISSUE #5261: Notification `iconDescription` is given to both the close icon and error/success/warning icon
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

Notification `iconDescription` is given to both the close icon and error/success/warning icon

> Is this issue related to a specific component?

`Toast/InlineNotification`

> What did you expect to happen? What happened instead? What would you like to
> see changed?

have separate props for the two icon descriptions? or only use them on the close button

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

latest

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1. go to https://react.carbondesignsystem.com/?path=/story/notifications--toast
1. switch to error/warning/success case
1. hover the icon on the left
1. hover the close icon
1. note they have the same text

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

bfa91880-4356-11ea-8053-3b20c2c5ab63

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/5261

Adds in a new prop, `statusIconDescription` that allows a user to add a description to the status icon. Defaults to `${kind} icon` if no prop is provided. 

#### Changelog

**New**

- `statusIconDescription`

**Removed**

- The status `NotificationIcon` no longer takes the same `iconDescription` as the close button

#### Testing / Reviewing

Ensure the status icon and close button have different text on hover, even if no description is provided.

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @tw15egan!

File: packages/react/src/components/Notification/Notification.js [hunk 1]
```diff
@@ -499,6 +506,11 @@ InlineNotification.propTypes = {
    */
   iconDescription: PropTypes.string.isRequired,
 
+  /**
+   * Provide a description for "status" icon that can be read by screen readers
+   */
+  statusIconDescription: PropTypes.string,
```
Comment (emyarod) [inline packages/react/src/components/Notification/Notification.js]:
```suggestion
  statusIconDescription: PropTypes.string.isRequired
```

does this also need to be required like `iconDescription`?

**Review by emyarod — APPROVED** [review]
looks good to me, just one question about the prop requirement

File: packages/react/src/components/Notification/Notification.js (hunk 1 above)
Comment (tw15egan) [reply packages/react/src/components/Notification/Notification.js]: Yeah makes sense to be consistent 👍

**Pull request merged**